test/unit/Schema.js

```diff
@@ -76,6 +76,10 @@ describe("Schema", () => {
 		expect(() => new dynamoose.Schema({"id": String, "friend": {"type": Object, "schema": {"name": {"type": String, "index": {"global": true}}}}})).to.throw("Index must be at root object and not nested in object or array.");
 	});
 
+	it("Should not throw error if passing multiple data types for a nested array attribute", () => {
```

Comment:
Just to confirm, this test fails without the changes you already made, correct?

Reply:
Yep, this test currently fails without the array check and passes with it.